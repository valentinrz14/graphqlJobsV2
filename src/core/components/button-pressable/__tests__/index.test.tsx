import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import { Text } from 'react-native';
import { ButtonPressable } from '../index';

describe('buttonPressable', () => {
  it('should render buttonPressable', () => {
    const handleOnPress = jest.fn();

    const { toJSON, getByTestId } = render(
      <ButtonPressable testID="pressable-test-id" onPress={handleOnPress}>
        <Text>ButtonPressable</Text>
      </ButtonPressable>,
    );
    const pressableTestId = getByTestId('pressable-test-id');
    expect(pressableTestId).toBeUndefined();
    fireEvent.press(pressableTestId);

    waitFor(() => {
      expect(handleOnPress).toHaveBeenCalledTimes(1);
    });

    expect(toJSON()).toMatchSnapshot();
  });
});
