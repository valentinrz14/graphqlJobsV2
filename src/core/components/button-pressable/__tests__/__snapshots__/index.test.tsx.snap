// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buttonPressable should render buttonPressable 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  <Text>
    ButtonPressable
  </Text>
</View>
`;
